src/unix/syscall.c

```diff
@@ -2506,6 +2525,19 @@ closure_function(0, 2, void, syscall_io_complete_cfn,
     syscall_return(t, rv);
 }
 
+closure_function(0, 1, status, hostname_done,
+                 buffer, b)
+{
+    hostname = b;
+
+    /* Remove trailing newline, if any. */
+    char last_char = *(char *)(buffer_end(hostname) - 1);
+    if (last_char == '\n')
```

Comment:
This could also check for a carriage return in case the file was written with CR-LF (e.g. Windows).

Reply:
OK, I modified this part so that any trailing CR and LF characters are removed.